extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/ResteasyStandaloneRecorder.java

```diff
@@ -125,9 +127,16 @@ public void run() {
                 });
             }
         }
-        if (hasClasspathResources) {
+        if (!knownPaths.isEmpty()) {
             router = router == null ? Router.router(vertx) : router;
-            router.route().handler(StaticHandler.create(META_INF_RESOURCES));
+            StaticHandler staticHandler = StaticHandler.create(META_INF_RESOURCES);
+            router.route().handler(ctx -> {
+                if (knownPaths.contains(ctx.request().path())) {
```

Comment:
Also note that `ctx.request().path()` is not normalized, is the `raw` request path, so for a resouce named: `file.txt` this path will match `/file.txt` but `/somedir/../file.txt` won't